src/reset.rs

```diff
@@ -40,11 +40,10 @@ impl From<esp_reset_reason_t> for ResetReason {
             esp_reset_reason_t_ESP_RST_PANIC => Self::Panic,
             esp_reset_reason_t_ESP_RST_INT_WDT => Self::InterruptWatchdog,
             esp_reset_reason_t_ESP_RST_POWERON => Self::PowerOn,
-            esp_reset_reason_t_ESP_RST_UNKNOWN => Self::Unknown,
             esp_reset_reason_t_ESP_RST_BROWNOUT => Self::Brownout,
             esp_reset_reason_t_ESP_RST_TASK_WDT => Self::TaskWatchdog,
             esp_reset_reason_t_ESP_RST_DEEPSLEEP => Self::DeepSleep,
-            _ => unreachable!(),
+            _ => Self::Unknown,
```

Comment:
I don't mind altering it so that `Self::Unknown` becomes a catch all.

But: you are not fixing the **root cause** why it panicked in the first place, isn't it?
- It didn't panic because of `esp_reset_reason_t_ESP_RST_UNKNOWN` because we have that branch
- It panicked because of _something else_

So... how about we map this _something else_ (whatever it is) in the `match` statement? I mean _in addition_ to the change you introduce here?

Reply:
OK. I'll look at the reset reason code and add it to the enum. @ivmarkov Did you mean this?

Reply:
Yes.